src/main/js/main_html5.js

```diff
@@ -883,27 +880,27 @@ require("../../../html5-common/js/utils/environment.js");
      * @public
      * @method OoyalaVideoWrapper#setAudio
      * @param {String} trackId - the ID of the audio track to activate
-     * @returns {array} - list of available audio streams
+     * @callback OoyalaVideoFactory#raiseAudioChange
      */
     this.setAudio = function(trackId) {
-      if (this.currentAudioId !== trackId) {
-        var audioTracks = _video.audioTracks;
-        if (audioTracks && audioTracks.length) { //if audioTracks exist
-
+      var audioTracks = _video.audioTracks;
+      if (audioTracks && audioTracks.length) { // if audioTracks exist
+        var currentAudio = _.find(audioTracks, function (track) {
+          return track.enabled;
+        });
+        var currentAudioId = currentAudio.id;
```

Comment:
Can we add a null check for currentAudio? It's possible that currentAudio is null if no track in audioTracks is enabled. We probably also want to consider the situation where we are setting audio with no previously enabled audio track so that we still enable the new track.